src/test/java/htsjdk/samtools/seekablestream/SeekableStreamGZIPinputStreamIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+
+/**
+ * Tests for the integration between SeekableStream and GZIPInputStream.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class SeekableStreamGZIPinputStreamIntegrationTest extends HtsjdkTest {
+    private static final String TEST_CHR = "1";
+    private static Collection<Allele> alleles = Arrays.asList(Allele.create("A", true), Allele.create("C"));
+    private static String RANDOM_ATTRIBUTE = "RD";
+
+
+    private static File createBgzipVcfsWithVariableSize(final int firstRecordAttributeLength, final int nSmallRecords) throws Exception {
+        final VariantContext longRecord = new VariantContextBuilder("long", TEST_CHR, 1, 1, alleles)
+                .attribute(RANDOM_ATTRIBUTE, generateRandomString(firstRecordAttributeLength))
+                .make();
+        final File tempFile = Files.createTempFile("test" + firstRecordAttributeLength + "_" + nSmallRecords, ".vcf.gz").toFile();
+        try (final VariantContextWriter writer = new VariantContextWriterBuilder()
+                .setOptions(VariantContextWriterBuilder.NO_OPTIONS)
+                .setOutputFile(tempFile).build()) {
+            writer.setHeader(createTestHeader()); // do not write the header
+            writer.add(longRecord);
+            for (int i = 2; i <= nSmallRecords + 1; i++) {
+                final VariantContext smallRecord = new VariantContextBuilder("short", TEST_CHR, i, i, alleles).attribute(RANDOM_ATTRIBUTE, ".").make();
+                writer.add(smallRecord);
+            }
+        }
+        return tempFile;
+    }
+
+
+    private static VCFHeader createTestHeader() {
+        final VCFHeader header = new VCFHeader();
+        header.addMetaDataLine(new VCFInfoHeaderLine(RANDOM_ATTRIBUTE, 1, VCFHeaderLineType.Character, "random string"));
+        return header;
+    }
+
+    private static String generateRandomString(final int length) {
+        final Random random = new Random(TestUtil.RANDOM_SEED);
+        final StringBuilder builder = new StringBuilder(length);
+        for (int i = 0; i < length; i++) {
+            // generate number from 0 to 9, and append to the builder
+            builder.append(random.nextInt(10));
+        }
+        return builder.toString();
+    }
+
+    @DataProvider
+    public Iterator<Object[]> compressedVcfsToTest() throws Exception {
+        final List<Object[]> data = new ArrayList<>();
+        final int nSmallRecords = 1000000;
+        for (int firstRecordLength = 1000; firstRecordLength <= 10000; firstRecordLength+=1000) {
+            data.add(new Object[]{createBgzipVcfsWithVariableSize(firstRecordLength, nSmallRecords), nSmallRecords+1});
+        }
+        return data.iterator();
+    }
+
+    @Test(dataProvider = "compressedVcfsToTest")
+    public void testWrappedSeekableStreamInGZIPinputStream(final File input, final long nLines) throws Exception {
+        try (final LineReader reader = new BufferedLineReader(new GZIPInputStream(new SeekableFileStream(input)))) {
```

Comment:
This test is very good, it appears to definitely fail when run without seekableStream.available(). Maybe you should assert that the file was actually written properly though, perhaps by asserting the lenght is consistent when you open a blockCompressedInputStream and compare the two?

Reply:
I think that it such a test is over-killing, because we already know how many lines should be. Anyway, I added a different test for check the consistency with a `BlockCompressedInputStream`, checking every byte returned by the stream.